packages/chat-headless/src/models/clients/ChatEventClient.ts

```diff
@@ -0,0 +1,55 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { MessageRequest, MessageResponse } from "@yext/chat-core";
+import { ChatClient } from "./ChatClient";
+
+/**
+ * An event-driven client for processing user message and provide responses
+ * by emitting and listening to events.
+ * 
+ * @public
+ */
+export interface ChatEventClient {
+  /**
+   * Initializes the client, using credentials and data in the provided message to setup a chat session.
+   * 
+   * @param messageResponse - The message response that initiated the handoff to the chat client.
+   */
+  init(messageResponse: MessageResponse): Promise<void>;
+  
+  /**
+   * Registers an event listener for a specified event.
+   * Supported events are:
+   * - `message`: A new message has been received.
+   * - `typing`: The agent is typing.
+   * - `close`: The chat session has been closed.
+   * 
+   * @param eventName - The name of the event to listen to.
+   * @param cb - The callback function to be executed when the event is triggered.
+   */
+  on(eventName: string, cb: (data: any) => void): void;
```

Comment:
any chance we could make the types for `on` and `emit` stronger? Doesn't have to be as fancy as `chat-core-aws-connect` with the types of the callbacks -- at least having `eventName` only support `message`, `typing` and `close` seems like it could be a good change.